src/express/src/routes/chain.ts

```diff
@@ -0,0 +1,38 @@
+import { Router, Request, Response } from 'express';
+import { ChainMetadataSchema, ZChainName } from '@hyperlane-xyz/sdk';
+import { validateBody, validateRequestParam } from '../middleware/validateRequest.js';
+import { ChainService } from '../services/chainService.js';
+
+export function createChainRouter(chainService: ChainService) {
+  const router = Router();
+
+  router.get(
+    '/:chain/metadata',
+    validateRequestParam('chain', ZChainName),
+    async (req: Request, res: Response) => {
+      const metadata = await chainService.getChainMetadata(req.params.chain);
+      res.json(metadata);
+    },
+  );
+
+  router.post(
+    '/:chain/metadata',
+    validateRequestParam('chain', ZChainName),
```

Comment:
you can just include `:chain` within the body. i dont see any advantage of using both request params and body

Reply:
I'll play the devil's advocate but technically speaking, following the REST spec this is the correct way of defining  the endpoint